src/Core/Settings.h

```diff
@@ -746,7 +746,13 @@ class IColumn;
     M(Seconds, wait_for_async_insert_timeout, DBMS_DEFAULT_LOCK_ACQUIRE_TIMEOUT_SEC, "Timeout for waiting for processing asynchronous insertion", 0) \
     M(UInt64, async_insert_max_data_size, 1000000, "Maximum size in bytes of unparsed data collected per query before being inserted", 0) \
     M(UInt64, async_insert_max_query_number, 450, "Maximum number of insert queries before being inserted", 0) \
-    M(Milliseconds, async_insert_busy_timeout_ms, 200, "Maximum time to wait before dumping collected data per query since the first data appeared", 0) \
+    M(Milliseconds, async_insert_busy_timeout_ms, 200, "Maximum time to wait before dumping collected data per query since the first data appeared. If auto-adjusting is enabled through enable_async_insert_adaptive_busy_timeout, the setting corresponds to the initial time to wait, and the following wait time will be auto-adjusted", 0) \
+    M(Milliseconds, async_insert_poll_timeout_ms, 50, "Timeout for polling data from asynchronous insert queue", 0) \
+    M(Bool, enable_async_insert_adaptive_busy_timeout, false, "Enable adaptive busy timeouts for asynchronous inserts", 0) \
+    M(Milliseconds, async_insert_busy_timeout_min_ms, 50, "Minimum value of an asynchronous insert timeout if auto-adjusting is enabled through enable_async_insert_adaptive_busy_timeout", 0) \
+    M(Milliseconds, async_insert_busy_timeout_max_ms, 1000, "Maximum value of an asynchronous insert timeout if auto-adjusting is enabled through enable_async_insert_adaptive_busy_timeout", 0) \
```

Comment:
shouldn't this be same as `async_insert_busy_timeout_ms`?
in the current implementation `async_insert_busy_timeout_ms` is the total time limit which will force a flush, i.e. maximum we can keep the entries before flushing.
Also, what if user changed `async_insert_busy_timeout_ms` to 2s? He would get errors how `async_insert_busy_timeout_max_ms` is not okay, and then he needs to figure out what it is and what value to set it to.

Why not put `async_insert_busy_timeout_ms` as alias to `async_insert_busy_timeout_max_ms` to keep backward compatibility?

Reply:
This sounds good to me. Is there a way to alias two settings (async_insert_busy_timeout_ms and async_insert_busy_timeout_max_ms) so that they are tied to the same value?

Reply:
TIL `ALIAS` expression, will do.